### PR TITLE
#443: Add realloc array function

### PR DESCRIPTION
**Pull request by m3g4d1v3r**
This pull request adds the realloc_array wrapper to the code base (it's inspired on alloc_array's implementation).
This is a Work In Progress. Please let me know if any fixes should be made.

Best regards.
m3g4d1v3r

### CODE REVIEW

**Comment by m3g4d1v3r** [comment]
Issue #443

File: src/memory.c [hunk 1]
```diff
@@ -136,6 +136,16 @@ alloc_array( size_t item_count, size_t item_size ) {
   return alloc_mem(item_count * item_size);
 }
 
+void *
+realloc_array( const void *mem, size_t item_count, size_t item_size ) {
+  #include <stdio.h>
```
Comment (goatshriek) [inline src/memory.c]: this include may be left over from testing? At any rate, if it is still needed it shouldn't be inside of this function.

File: src/element.c [hunk 2]
```diff
@@ -56,18 +56,18 @@ struct stumpless_element *
 stumpless_add_param( struct stumpless_element *element,
                      struct stumpless_param *param ) {
   struct stumpless_param **new_params;
-  size_t old_params_size;
-  size_t new_params_size;
+  size_t old_params_count;
```
Comment (goatshriek) [inline src/element.c]: it appears that you can get rid of this local variable now, as the size calculation code that used it no longer needs it.

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thanks very much for putting this together! Just a few minor cleanups noted here and I believe it will be ready to merge.

File: src/memory.c (hunk 1 above)
Comment (m3g4d1v3r) [reply src/memory.c]: I'm sorry about this: I've really forgot to remove this header.

I've just refactored the 3 commits of this PR. Everything should be ok now.

File: src/element.c (hunk 2 above)
Comment (m3g4d1v3r) [reply src/element.c]: Ok: I've removed these variables and the "+ 1" modification that went to new_params_size (let me know if this change is what is expected).

I've just refactored the 3 commits of this PR by doing git rebase.

Comment (goatshriek) [reply src/element.c]: The plus one is needed to reallocate the array with space for the new param, which is probably why some of the tests are failing. You can probably just use `element->param_count +1` as the second argument to `realloc_array` to fix this.

Comment (m3g4d1v3r) [reply src/element.c]: Hello! 

Right: I've just changed the second argument of this `realloc_array` call to `param_count + 1`.

**Review by goatshriek — APPROVED** [review]

**Pull request merged**